Foundation/ScannerAPI.swift

```diff
@@ -0,0 +1,198 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+extension CharacterSet {
+    fileprivate func contains(_ character: Character) -> Bool {
+        return character.unicodeScalars.allSatisfy(self.contains(_:))
+    }
+}
+
+// -----
+
+@available(swift 5.0)
+extension Scanner {
+    public enum NumberRepresentation {
+        case decimal // See the %d, %f and %F format conversions.
+        case hexadecimal // See the %x, %X, %a and %A format conversions. For integers, a leading 0x or 0X is optional; for floating-point numbers, it is required.
+    }
+    
+    public var currentIndex: String.Index {
+        get {
+            let string = self.string
+            var index = string._toUTF16Index(scanLocation)
+            
+            var delta = 0
+            while index != string.endIndex && index.samePosition(in: string) == nil {
+                delta += 1
+                index = string._toUTF16Index(scanLocation + delta)
+            }
+            
+            return index
+        }
+        set { scanLocation = string._toUTF16Offset(newValue) }
+    }
+    
+    public func scanInt(representation: NumberRepresentation = .decimal) -> Int? {
+        #if arch(x86_64) || arch(arm64) || arch(s390x) || arch(powerpc64) || arch(powerpc64le)
```

Comment:
Would it be better to write this as `if Int.bitWidth == Int64.bitWidth` to avoid having to maintain a list of archs?

Reply:
We use this in `Data` as well just to save the runtime check.

Reply:
Wouldnt it be a constant expression and optimised away by the compiler?

Reply:
There were issues with that IIRC, and it may or may not be.